src/Orleans/Core/GrainExtensions.cs

```diff
@@ -223,6 +228,20 @@ public static string GetPrimaryKeyString(this IGrainWithStringKey grain)
                 throw new InvalidOperationException("Invoking of methods with one way flag must result in completed task");
             }
         }
+        
+        public static IStorageProvider GetStorageProvider(this Grain grain, IServiceProvider services)
```

Comment:
At the very least I would constrain it to `Grain<T>`, not every `Grain`, but I think I prefer in something different than GrainExtensions that doesn't pollute the intellisense (or even not provide an extension method for this).
As you mentioned it's probably only going to be used by some infrastructure code (even just from us). And as we endorse facets, this is even less relevant.

Reply:
probably missed addressing this one

Reply:
Didn't fix because there was nothing broken.  Moved to namespace Orleans.Storage anyway, so it won't show up unless someone is already using storage features.